File: web/src/app/core/realtime/realtime.events.ts
// Mirrors src/TicketSystem.Application/Features/Realtime/RealtimeEvents.cs
export interface TicketCreatedEvent {
  ticketId: string;
  customerId: string;
  assignedAgentId: string | null;
}

export interface TicketUpdatedEvent {
  ticketId: string;
  changeKind: 'Status' | 'Priority' | 'Closed';
  customerId: string;
  assignedAgentId: string | null;
}

export interface TicketAssignedEvent {
  ticketId: string;
  oldAssigneeId: string | null;
  newAssigneeId: string | null;
  customerId: string;
}

export interface CommentAddedEvent {
  ticketId: string;
  commentId: string;
}

export interface TimeEntryLoggedEvent {
  ticketId: string;
  timeEntryId: string;
}

File: src/TicketSystem.Application/Features/Realtime/RealtimeEvents.cs
namespace TicketSystem.Application.Features.Realtime;

/// <summary>
/// Small typed events broadcast by the SignalR hub. Deliberately minimal — no display
/// fields; clients refetch the authoritative REST endpoint on any event. Reduces coupling
/// between write handlers and client render shape.
/// </summary>
public sealed record TicketCreatedEvent(Guid TicketId, Guid CustomerId, Guid? AssignedAgentId);

public sealed record TicketUpdatedEvent(
    Guid TicketId,
    string ChangeKind,          // "Status" | "Priority" | "Closed"
    Guid CustomerId,
    Guid? AssignedAgentId);

public sealed record TicketAssignedEvent(
    Guid TicketId,
    Guid? OldAssigneeId,
    Guid? NewAssigneeId,
    Guid CustomerId);

public sealed record CommentAddedEvent(Guid TicketId, Guid CommentId);

public sealed record TimeEntryLoggedEvent(Guid TicketId, Guid TimeEntryId);
